### PR TITLE
bf: append authdata accounts to existing accounts

### PR DESCRIPTION
**Pull request by philipyoo**
Instead of overwriting existing accounts, append new accounts. This is needed to support file backend locally

### CODE REVIEW

**Comment by RachedBenMustapha** [comment]
@philipyoo it seems to me that when new configuration versions are applied, all accounts are appended, so adding 3 orbit users (3 configuration versions) the user list would end up like so: [fileuser1, fileuser2, orbituser1, orbituser1, orbituser2, orbituser1, orbituser2, orbituser3] instead of [fileuser1, fileuser2, orbituser1, orbituser2, orbituser3]. Did you get a chance to test that case?

Maybe a filtering out of users with source "orbit" (could be a new attribute) before the concat would solve this problem.

File: lib/Config.js [hunk 1]
```diff
@@ -966,7 +969,20 @@ class Config extends EventEmitter {
     }
 
     setAuthDataAccounts(accounts) {
-        this.authData.accounts = accounts;
+        // append orbit's users instead of replacing existing accounts
+        let allAccounts = this.authData.accounts;
+        if (accounts.every(a => a.source === 'orbit')) {
```
[the remark this answers is not in the document]
Comment (philipyoo) [reply lib/Config.js]: ah nice!

**Comment by RachedBenMustapha** [comment]
@philipyoo one last thing, the default accounts in the authdata file have weak and hardcoded keys so the default behavior should be to exclude them and have an option to keep them around. Otherwise it would be a pretty big security concern.

**Comment by philipyoo** [comment]
@RachedBenMustapha Good point! I'm thinking if the accounts from orbit contain the default `arn` and `shortid`, then to append to the existing `authdata.json` profiles. Otherwise, to overwrite the `this.authData.accounts` in Configs?
I'm not sure if that is a valid check to determine this though

**Comment by rahulreddy** [comment]
@philipyoo Can you use a Jira for this and tag the branch/commits accordingly?

**Pull request closed without merging**